[frio] Better navigation

xundeenergie opened the pull request:
Make a few changes to have shortcut-buttons from profile to create posting with mentioning this user/group and to switch directly to the group, when viewing the contact of a group.

Described here
https://soc.schuerz.at/display/4edd2508-1365-19a2-9c65-218205372300

Commit:
Frio: open links to source in a new browser tab

Commit:
add "Show Forum" Button to Profile

Commit:
add mention-buttons

Commit:
Change label of mention-button

* "Mention" on users profile
* "Post to group" on group/forums profile

Commit:
set label in hovercard for mentionings

Commit:
show "view group" in hovercard

but only on group-contacts.
rename all *forum to *group in code

Commit:
changes to fit to showgroup-button branch

Commit:
set ! and @ on mentioning in hover-card

Commit:
remove mention-button from own profile

Commit:
set variables for correct mentioning

Commit:
add translations for de for mention-buttons

Commit:
add translation de for showgroup button

Commit:
Merge branch 'add-mention-button' into better-navigation

Commit:
Merge branch 'showgroup-button' into better-navigation

Commit:
fix Typo

Commit:
regenerate translations

MrPetovan committed:
Allow macOS users to use Ctrl + Cmd + Space to bring up the emoji picker in Friendica text fields

- Remove Backspace guard against changing page
- Clarify keyboard shortcut to pause automatic updates

MrPetovan commented on view/lang/de/strings.php:
Please remove your translations for specific languages from your pull request, they will be clobbered by the next import from Transifex anyway.

xundeenergie replied:
hmmm you told me https://friendica.mrpetovan.com/display/735a2029-1865-19c5-d3db-ced198096018

what exactly should i remove? which files?

view/lang/C/messages.po?

MrPetovan replied:
No, this one is fine, but the changes to `view/lang/de/strings.php` shouldn't be included in this PR.

xundeenergie replied:
Ah. Ok I understand.

xundeenergie replied:
done.
removed my translations and run bin/run_xgettext.sh again

Commit:
remove specific translation

Commit:
regenerate xgettext

MrPetovan commented on src/Content/Widget/VCard.php:
```diff
@@ -99,6 +100,10 @@ public static function getHTML(array $contact): string
 				$wallmessage_link = 'message/new/' . $id;
 			}
+
+			if (in_array($rel, [Contact::SHARING])) {
```
Are you specifically testing whether the contact is a group here? If yes, this condition seems too wide, it would match all the contacts you follow.

xundeenergie replied:
yes that's right. I'm testing if a contact is a group... 
how can i test this faster and better in this place?

MrPetovan replied:
```suggestion
			if ($contact['contact-type'] == Contact::TYPE_COMMUNITY) {
```

xundeenergie replied:
thank you!
changed it.

MrPetovan commented on src/Content/Item.php:
```diff
@@ -432,6 +432,8 @@ public function photoMenu(array $item, string $formSecurityToken): string
 				$this->l10n->t('View Contact') => $contact_url,
 				$this->l10n->t('Send PM') => $pm_url,
+				$this->l10n->t('Mention') => $mention_url,
+				$this->l10n->t('Post to group') => $mention_url,
```
Why is it the same URL in both cases?

xundeenergie replied:
I must confess, i don't know, what happens here... 
Have orientated on the code... but now i removed this lines completely, and it works also.

Can you please explain, what this function does?

MrPetovan commented on src/Content/Widget/VCard.php:
```diff
@@ -119,6 +124,10 @@ public static function getHTML(array $contact): string
 			'$unfollow_link'    => $unfollow_link,
 			'$wallmessage'      => DI::l10n()->t('Message'),
 			'$wallmessage_link' => $wallmessage_link,
+			'$mentioning'       => DI::l10n()->t('Mention'),
+			'$post2group'       => DI::l10n()->t('Post to group'),
```
Please rename closer to the target label.
```suggestion
			'$mention'          => DI::l10n()->t('Mention'),
			'$posttogroup'      => DI::l10n()->t('Post to group'),
```

xundeenergie replied:
changed it.

MrPetovan commented on src/Model/Contact.php:
```diff
@@ -1197,8 +1199,20 @@ public static function photoMenu(array $contact, int $uid): array
 			$pm_url = 'message/new/' . $contact['id'];
 		}
 
+		if (in_array($contact['rel'], [contact::SHARING])) {
```
Same question about targeting groups, this would match all the contacts you are following.

MrPetovan replied:
```suggestion
		if ($contact['contact-type'] == self::TYPE_COMMUNITY) {
```

xundeenergie replied:
changed the code to your suggestion. it works.

MrPetovan commented on src/Model/Contact.php:
```diff
@@ -1197,8 +1199,20 @@ public static function photoMenu(array $contact, int $uid): array
+		}
+
+
+		if (in_array($contact['rel'], [contact::SHARING])) {
+			$self = true;
+		}
```
What is the meaning of this? It's the same condition as above, but for yet another purpose?

tobiasd committed:
Updated the translations

Raroun committed:
added new tab for wall-titles too

Raroun committed:
added missing quotation mark

Commit:
fix code because of review

MrPetovan commented on src/Model/Contact.php:
```diff
@@ -1232,6 +1244,9 @@ public static function photoMenu(array $contact, int $uid): array
 				'unfollow' => [DI::l10n()->t('Unfollow'), $unfollow_link, true],
+				'mention'  => [DI::l10n()->t('Mention'), DI::l10n()->t('Post to group'), $mention_url, false],
+				'group'    => [DI::l10n()->t('View group'), $group_link, $contact['forum'], true],
+				'self'     => [$contact['self'] ?? false, true],
```
Why does this template variable have two potentially different values given that you only query the first in the template?
```suggestion
				'self'     => $contact['self'] ?? false,
```
This would also be better placed in  `src/Content/Widget/Hovercard.php` under the `'$profile'` template variable.

xundeenergie replied:
I don't know why, but if i set it only with one value, it is empty, when i query it's content...

MrPetovan replied:
~~Since it's a boolean value, it shouldn't produce any output when you try to display it in the template.~~

Edit: It's actually an integer that can be 0 or 1, so it should have displayed in the template.

MrPetovan commented on view/templates/hovercard.tpl:
```diff
@@ -22,8 +22,12 @@
 				<div class="hover-card-actions-social">
 					{{if $profile.actions.pm}}<a class="btn btn-labeled btn-primary btn-sm add-to-modal" href="{{$profile.actions.pm.1}}" aria-label="{{$profile.actions.pm.0}}"><i class="fa fa-envelope" aria-hidden="true" title="{{$profile.actions.pm.0}}"></i><span class="sr-only">{{$profile.actions.pm.0}}</span></a>{{/if}}
+
+					{{if $profile.addr && !$profile.actions.self.0}}<a class="btn btn-labeled btn-primary btn-sm" href="{{$profile.actions.mention.2}}" aria-label="{{if $profile.actions.group.2==1}}{{$profile.actions.mention.1}}{{else}}{{$profile.actions.mention.0}}{{/if}}" title="{{if $profile.actions.group.2==1}}{{$profile.actions.mention.1}}{{else}}{{$profile.actions.mention.0}}{{/if}}"><i class="fa fa-penci ...
```
After you moved the `self` key to the Hovercard template variable.
```suggestion
					{{if $profile.addr && !$profile.self}}<a class="btn btn-labeled btn-primary btn-sm" href="{{$profile.actions.mention.2}}" aria-label="{{if $profile.actions.group.2==1}}{{$profile.actions.mention.1}}{{else}}{{$profile.actions.mention.0}}{{/if}}" title="{{if $profile.actions.group.2==1}}{{$profile.actions.mention.1}}{{else}}{{$profile.actions.mention.0}}{{/if}}"><i class="fa fa-pencil-square-o" aria-hidden="true"></i></a>{{/if}}
```

xundeenergie replied:
> After you moved the `self` key to the Hovercard template variable.

you mean `'self'     => $contact['self'] ?? false,` instead of `'self'     => [$contact['self'] ?? false, true],`?

I tried this before i pushed the commit. It did not work.
And i tried it again, the "mention"-button is shown on my own profiles hovercard.
With my solution, it is not shown.

Or am i doing it wrong?

xundeenergie replied:
I don't understand this... 
The only solution, which shows the mention-button on others and hide it on my own profile is the following:

in Contact.php 
`'self'     => [$self, true],`  or `'self'     => [$contact['self'] ?? false, true],` in both `$menu = [` parts (where $self is 0 or 1 or its true or false) 
and in hovercard.tpl
`!$profile.actions.self.0` in the if-clause... 

i played around with all possible combinations... nothing else works (event not, what you suggested).
Would be glad, if you can explain this. I really want to understand what happens here.

MrPetovan replied:
I'll pull your branch, see what I can do with it.

MrPetovan replied:
I was able to enable my changes and I couldn't resist simplifying the template further by moving group-based conditions to the PHP logic in `Model\Contact::photoMenu`. If you want to keep working on your branch, be sure to pull it first.

xundeenergie replied:
> I was able to enable my changes and I couldn't resist simplifying the template further by moving group-based conditions to the PHP logic in `Model\Contact::photoMenu`. If you want to keep working on your branch, be sure to pull it first.

yes, i will do.
but which branch from where i should pull?

xundeenergie replied:
i see!

Did you remove the "View group"-button?
Or is it just removed, because i should reimplement it using the new logic?

xundeenergie replied:
I see the new logic, and improved it a bit, to show the group-icon instead of cloud-icon for "Show group" in hovercard for groups.

MrPetovan replied:
Did you pull your branch first before committing your new changes? They don't show in the PR yet.

xundeenergie replied:
Ok... i did...

Will check it at home again.

FarisKarim committed:
add lazy loading for images

MrPetovan committed:
Move self key to $profile tamplate variable in hovercard.tpl template

- Simplify network and mention menu items by checking for group in Model\Contact::photoMenu

Commit:
add self for unfollowed contact

Commit:
Merge remote-tracking branch 'upstream' into better-navigation

Commit:
improve button-icon for groups in hovercard

xundeenergie commented:
Tank you 😊

MrPetovan commented:
No, thank *you* for your work!
